src/Game/Game.elm

```diff
@@ -103,8 +105,12 @@ update msg model =
                 model' =
                     Game.Collision.tryMoveHero dir model
 
+                (viewportX, viewportY) = updateViewportOffset model.hero.position model'
+
+                model'' = { model' | viewportX = viewportX, viewportY = viewportY }
```

Comment:
Almost there, please move this into `updateViewportOffset : Vector -> Model -> Model`. 

In elm, the convention is you can return a model and now, I typically name the model like this `model' -> modelWithMovedHero` and `model'' -> modelWithMovedHeroUpdatedViewport` to give some context.